packages/salesforcedx-vscode-core/src/commands/forceApexTestRunCodeAction.ts

```diff
@@ -34,15 +72,82 @@ export class ForceApexTestRunCodeActionExecutor extends SfdxCommandletExecutor<{
       .withFlag('--tests', this.test)
       .withFlag('--resultformat', 'human')
       .withArg('--synchronous')
+      .withFlag('--loglevel', 'error')
       .build();
   }
 }
 
-export function forceApexTestRunCodeAction(test: string) {
+function forceApexTestRunCodeAction(test: string) {
   const commandlet = new SfdxCommandlet(
     new SfdxWorkspaceChecker(),
     new EmptyParametersGatherer(),
     new ForceApexTestRunCodeActionExecutor(test)
   );
   commandlet.run();
 }
+
+//   T E S T   C L A S S
+
+// evaluate test class param: if not provided, apply cached value
+// exported for testability
+export function resolveTestClassParam(testClass: string): string {
+  const testCache = ForceApexTestRunCache.getInstance();
+  if (isEmpty(testClass)) {
+    // value not provided for re-run invocations
+    // apply cached value, if available
+    if (testCache.hasCachedClassTestParam()) {
+      testClass = testCache.lastClassTestParam;
+    }
+  } else {
+    testCache.lastClassTestParam = testClass;
+  }
+
+  return testClass;
+}
+
+// invokes apex test run on all tests in a class
+export function forceApexTestClassRunCodeAction(testClass: string) {
```

Comment:
Do you have a screenshot of this? How does the user know that she is actually an older class/method? Does the name show up prominently in the UI? Or is it implicitly assumed?

Reply:
You mean on re-run?

We can add the test class or test method name to the command title in apex/jorje/lsp/impl/codeActions/ApexTestRunCodeActionProvider.java.  Would that be better?